joplin/importer/page_importer.py

```diff
@@ -16,18 +20,65 @@
 }
 
 
-def change_keys(obj, convert):
+def change_keys(obj, convert, page_dictionary):
     """
     Recursively goes through the dictionary obj and replaces keys with the convert function.
     """
-    if isinstance(obj, (str, int, float)):
+    if isinstance(obj, (int, float)):
         return obj
+    if isinstance(obj, str):
+        # if our string is html and has links, check for/recreate internal links
+        soup = BeautifulSoup(obj, 'html.parser')
+        if soup and soup.find('a'):
+            # get all the links
+            for link in soup.find_all('a'):
+                # get a urllib.parse result to play with
+                parse_result = urlparse(link.get('href'))
+
+                # check by hostname to see if this isn't an internal link
+                if parse_result.hostname not in ENDPOINTS:
```

Comment:
Our ENDPOINTS logic might have to be updated in the future to accommodate PRs. Not an issue right now though